src/element/textElement.ts

```diff
@@ -28,6 +28,7 @@ import {
   resetOriginalContainerCache,
   updateOriginalContainerCache,
 } from "./textWysiwyg";
+import { mockMeasureText } from "../tests/test-utils";
```

Comment:
wait, this would mean we're bundling test-utils and who knows what else.

Reply:
Good point!
`mockMeasureText` will anyways get bundled unless dynamically imported which doesn't make sense here, probably then moving `measureText`to a separate file ?
And we are doing all these just to make mocks work :D

Reply:
No, let's not split/rewrite our files just to satisfy tests.

One hack is to reimport the file inside itself:

```
// src/element/textElement.ts
import * as thisModule from "./textElement";
```

and call the `measureText` using `thisModule.measureText()`.

But this might constitute a circular dependency which will disallow later.

---

Also, completely mocking out the implementation of a function (or its significant part) kinda defeats the purpose of the test to begin with, no?

Rather, we can mock the DOM API we're using for dimension calculation that's not working in jest.

Reply:
Mocking DOM api is unnecessary here since that's not something we need to test, we trust the DOM API. what is needed here is `measureText` returning some measurements instead of 0/1 values and which should change on different text/font etc, hence I was mocking only that part.

Reply:
I have skipped the test for now. Will look for better mock and fix it later.